### ISSUE TITLE
HTML validation error due to missing aria-level for <p> submenu captions

### ISSUE THREAD

**Issue by jobovy** [sphinx-doc/sphinx#14030]
### Describe the bug

The change introduced in https://github.com/sphinx-doc/sphinx/pull/9358 to better support screen readers by adding the "heading" role to TOC captions is now leading to HTML validation errors because ``<p role="heading">`` also need to have aria-level set (e.g., HTML validator update in https://github.com/validator/validator/pull/1751 that now requires this).

I'm not quite sure what the correct aria-level is in general, that is, whether it would ever not be 1? If it's always 1 or always some other number, it would be a straightforward change.

### How to Reproduce

Use a .. toctree:: with a :caption:

### Environment Information

```text
Platform:              darwin; (macOS-15.7.1-arm64-arm-64bit)
Python version:        3.12.10 | packaged by conda-forge | (main, Apr 10 2025, 22:19:24) [Clang 18.1.8 ])
Python implementation: CPython
Sphinx version:        8.1.3
Docutils version:      0.21.2
Jinja2 version:        3.1.6
Pygments version:      2.19.1
```

### Sphinx extensions

```python

```
